### PR TITLE
Fix IASL disappear every 2nd time building.

### PR DESCRIPTION
**Pull request by PhilXing**
Fix IASL disappear every 2nd time building, while replace some paths variable with exist definitions.

### CODE REVIEW

**Comment by PhilXing** [comment]
The building process will fail every 2nd time caused by IASL folder removed, fixed by removing the redundant process within _check_iasl.

File: Makefile [hunk 1]
```diff
@@ -318,7 +309,7 @@ ifeq ($(and $(FW_VER_LT_1_6),$(FW_VER_NE_0_00)),)
 	@ln -sf $(OUTPUT_IMAGE) $(TIANOCORE_ATF_IMAGE)
 else
 	$(eval TIANOCORE_ATF_SIGNED_IMAGE := $(WORKSPACE)/Build/$(BOARD_NAME_UFL)/$(BUILD_VARIANT)_$(EDK2_GCC_TAG)/$(BOARD_NAME)_tianocore_atf.img.signed)
-	$(eval DBU_KEY := $(EDK2_PLATFORMS_SRC_DIR)/Platform/Ampere/$(BOARD_NAME_UFL)Pkg/TestKeys/Dbu_AmpereTest.priv.pem)
+	$(eval DBU_KEY := $(EDK2_PLATFORMS_PKG_DIR)/TestKeys/Dbu_AmpereTest.priv.pem)
```
Comment (thcao-ampere) [inline Makefile:312]: Why did you change this location ? Ampere put keys to Platform/Ampere/JadePkg/TestKeys

File: Makefile [hunk 2]
```diff
@@ -257,7 +248,7 @@ _tianocore_prepare: _check_source _check_tools _check_compiler _check_iasl
 
 _tianocore_sign_fd: _check_atf_tools
 	@echo "Creating certitficate for $(OUTPUT_FD_IMAGE)"
-	$(eval DBB_KEY := $(EDK2_PLATFORMS_SRC_DIR)/Platform/Ampere/$(BOARD_NAME_UFL)Pkg/TestKeys/Dbb_AmpereTest.priv.pem)
+	$(eval DBB_KEY := $(EDK2_PLATFORMS_PKG_DIR)/TestKeys/Dbb_AmpereTest.priv.pem)
```
Comment (thcao-ampere) [inline Makefile:251]: Why did you change this location ? Ampere put keys to Platform/Ampere/JadePkg/TestKeys

File: Makefile (hunk 1 above)
Comment (PhilXing) [reply Makefile:312]: I did not change the location, EDK2_PLATFORMS_PKG_DIR equal to EDK2_PLATFORMS_SRC_DIR)/Platform/Ampere/$(BOARD_NAME_UFL)Pkg according to line 28.
Changing this expression makes me easier to replace the keys via override EDK2_PLATFORMS_PKG_DIR, but that would be the next modification to modify line 28 replacing ":=" with "?=".

Comment (thcao-ampere) [reply Makefile:312]: got it 

I think don't need to change ":=" with "?=" - it's defined with EDK2_PLATFORMS_SRC_DIR

Comment (PhilXing) [reply Makefile:312]: Because I cannot submit my platform code into "edk2-platforms" (=EDK2_PLATFORMS_SRC_DIR) due to company IP policy, so I make it a submodule of my project now, and the BOARD_NAME is different from "Jade", I have to replace them via changing EDK2_PLATFORMS_PKG_DIR. I think most OEM will encounter the similar situation because I have not seen any platform code other than "Jade".

Comment (thcao-ampere) [reply Makefile:312]: With ":=", you still pass it when you compile

`make EDK2_PLATFORMS_PKG_DIR=<your path> .....`

Comment (PhilXing) [reply Makefile:312]: you are right, I will cease the modification, thanks for the tip!

File: Makefile [hunk 3]
```diff
@@ -192,13 +192,6 @@ _check_iasl:
 	@echo -n "Checking iasl..."
 	$(eval IASL_NAME := acpica-unix2-$(IASL_VER))
 	$(eval IASL_URL := "https://acpica.org/sites/acpica/files/$(IASL_NAME).tar.gz")
-ifneq ($(shell $(IASL) -v 2>/dev/null | grep $(IASL_VER)),)
-# iASL compiler is already available in the system.
-	@rm -rf $(IASL_DIR)
-	@echo "OK"
-else
```
Comment (vu-ampere) [inline Makefile:199]: Hi Phil, this just remove the downloaded iasl when the matching version is available in the system. Can you show me how you met this issue ?

Comment (PhilXing) [reply Makefile:199]: I cannot duplicate it now, there must be something else I have touched other than makefile to cause this issue, please ignore this PR if you cannot duplicate either via 
making the platform "JadePkg" twice in the row, sorry about troubling you...

Comment (vu-ampere) [reply Makefile:199]: No problem, just open a new one if you are able to reproduce it. Anyway, I find other changings still have their value.

Comment (PhilXing) [reply Makefile:199]: I think it might be caused by "edk2-platforms.config": by the time that I had not override this file path, no OEM section within the one in the "edk2-ampere-tools" either, so no "IASL_VER" could be parsed, which caused this issue. but debugging this issue made by myself is not priority task, just for your reference.

Comment (thcao-ampere) [reply Makefile:199]: If you want to change isal version, you need pass this variable in your make command

`# make IASL_VER=<iasl_version> ...`

Comment (philxingadlink) [reply Makefile:199]: Thanks for the tip!
BTW, if modification other than the "IASL_VER" related can be merged, it would be very helpful for me to make this repository a submodule of my project.

Comment (PhilXing) [reply Makefile:199]: I can duplicate while the make target is "all", please help to confirm; I make BIOS like this:
  make -f $WORKSPACE/edk2-ampere-tools/Makefile \
      BOARD_SETTING=$WORKSPACE/srp-ac01-fw-aptv-bin-r1.06.20210528/altra_firmware_sdk/bin/board_settings/jade_board_setting_1.06.20210526.bin \
      ATF_SLIM=$WORKSPACE/srp-ac01-fw-aptv-bin-r1.06.20210528/altra_firmware_sdk/bin/atf/altra_atf_signed_1.06.20210526.slim \
      LINUXBOOT_BIN=$WORKSPACE/flashkernel \
      all
